app/controllers/dashboard_controller.rb

```diff
@@ -10,6 +10,45 @@ def index
     end
   end
 
+  def create
+    Dashboard::Manager.add_widget_to_user(User.current, params[:widget])
+    redirect_to root_path
+  end
+
+  def destroy
+    widget = Widget.find(params[:id])
+    if widget.present? and widget.user == User.current
```

Comment:
Only the user right now, since there isn't a way for the admin to mange users' dashboards. 
If we create a dashboard management interface for admins, this should change, but right now I don't think we're there yet.